src/asn1acn.cpp

```diff
@@ -240,6 +252,35 @@ void Asn1AcnPlugin::addBuildICDToProjectMenu()
     menu->addAction(command, ProjectExplorer::Constants::G_PROJECT_FILES);
 }
 
+void Asn1AcnPlugin::initializeGenerateTestsAction(ActionContainer *toolsMenu)
+{
+    auto action = new Utils::ParameterAction(tr("Generate Tests..."),
+                                             tr("Generate Tests for \"%1\"..."),
+                                             Utils::ParameterAction::AlwaysEnabled,
+                                             this);
+    action->setEnabled(false);
+
+    connect(ProjectExplorer::SessionManager::instance(),
+            &ProjectExplorer::SessionManager::startupProjectChanged,
+            [action](ProjectExplorer::Project *project) {
+                action->setEnabled(project != nullptr);
+                action->setParameter(project == nullptr ? QString() : project->displayName());
+            });
+
+    connect(action, &QAction::triggered, [this]() {
+        if (m_testGeneratorDialog.isNull()) {
+            auto paramsProvider = std::make_shared<TestGenerator::TestGeneratorParamsProvider>(
+                Settings::load<Settings::TestGenerator>());
```

Comment:
maybe it should get settings loaded earlier (in function argument)?

Reply:
I will fix this